### ISSUE TITLE
fastlane create new certificate for new project instead of using existing one

### ISSUE THREAD

**Issue by gtatgh** [fastlane/fastlane#9865]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

fastlane (sigh) created new certificate for new project instead of use existing one. I have an existing project which uses old certificate. Both projects are configured to use same accounts and team id. The difference is the old project was configured by Appfile and Matchfile and new one uses .env file. Now when I compile existing project, I'm getting following error.
**Edit 2017-07-26**
I have tried force update PP for old project but without success.

##### Complete output when running fastlane, including the stack trace and command used
<details><summary>Error</summary>

[16:58:36]: ▸ ❌  Provisioning profile "match AdHoc com.company.app.project.watchkitapp.watchkitextension" doesn't include signing certificate "iPhone Distribution: company City GmbH (TEAM_ID)".

[16:58:36]: ▸ ❌  Code signing is required for product type 'WatchKit Extension' in SDK 'watchOS 3.2'

[16:58:36]: ▸ ** ARCHIVE FAILED **

[16:58:36]: ▸ The following build commands failed:

[16:58:36]: ▸ Check dependencies

[16:58:36]: ▸ (1 failure)
</details>
<details><summary>Creating new certificate and provisioning profiles</summary>

[14:31:06]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile

[14:31:08]: -------------------------------------------------

[14:31:08]: --- Step: Verifying required fastlane version ---

[14:31:08]: -------------------------------------------------

[14:31:08]: Your fastlane version 2.48.0 matches the minimum requirement of 2.47.0  ✅

[14:31:08]: ------------------------------

[14:31:08]: --- Step: default_platform ---

[14:31:08]: ------------------------------

[14:31:08]: Driving the lane 'ios setup' 🚀

[14:31:08]: Lane setup started.

[14:31:08]: ----------------------------------------------

[14:31:08]: --- Step: Switch to ios create_app_id lane ---

[14:31:08]: ----------------------------------------------

[14:31:08]: Cruising over to lane 'ios create_app_id' 🚖

[14:31:08]: ---------------------

[14:31:08]: --- Step: produce ---

[14:31:08]: ---------------------

+----------------+-------------------------------------+
|              Summary for produce 2.48.0              |
+----------------+-------------------------------------+
| app_identifier | com.company.app.NewProject |
| app_name       | NewProject                    |
| username       | iosdev@company.com           |
| sku            | 12345                          |
| platform       | ios                                 |
| language       | English                             |
| skip_itc       | true                                |
| skip_devcenter | false                               |
| team_id        | TEAMID                          |
| team_name      | company City GmbH           |
+----------------+-------------------------------------+

[14:31:11]: Creating new app 'NewProject' on the Apple Dev Center
[14:31:12]: Created app XYZ
[14:31:12]: Finished creating new app 'NewProject' on the Dev Center
[14:31:12]: Cruising back to lane 'ios setup' 🚘
[14:31:12]: ---------------------------------------------
[14:31:12]: --- Step: Switch to ios sync_signing lane ---
[14:31:12]: ---------------------------------------------
[14:31:12]: Cruising over to lane 'ios sync_signing' 🚖
[14:31:12]: -------------------
[14:31:12]: --- Step: match ---
[14:31:12]: -------------------

+-----------------------+-----------------------------------------------+
|                       Summary for match 2.48.0                        |
+-----------------------+-----------------------------------------------+
| git_url               | ssh://Bamboo-Daemon@xyz-yxz-stash:7999/gad  |
|                       | /ios_certificates.git                         |
| git_branch            | com.company.app.NewProject           |
| type                  | adhoc                                         |
| app_identifier        | com.company.app.NewProject           |
| username              | iosdev@company.com                     |
| keychain_name         | login.keychain                                |
| readonly              | false                                         |
| team_id               | TEAMID                                    |
| team_name             | company City GmbH                     |
| verbose               | false                                         |
| force                 | false                                         |
| skip_confirmation     | false                                         |
| shallow_clone         | false                                         |
| clone_branch_directly | false                                         |
| force_for_new_devices | true                                          |
| skip_docs             | false                                         |
| platform              | ios                                           |
+-----------------------+-----------------------------------------------+

[14:31:12]: Cloning remote git repo...
[14:31:12]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
[14:31:13]: Checking out branch com.company.app.NewProject...
[14:31:13]: 🔓  Successfully decrypted certificates repo
[14:31:13]: Verifying that the certificate and profile are still valid on the Dev Portal...
[14:31:15]: Couldn't find a valid code signing identity in the git repo for distribution... creating one for you now

+---------------+-----------------------------------------------+
|                    Summary for cert 2.48.0                    |
+---------------+-----------------------------------------------+
| development   | false                                         |
| force         | true                                          |
| username      | iosdev@company.com                     |
| team_id       | TEAMID                                    |
| keychain_path | /Users/user/Library/Keychains/login.keychai  |
|               | n-db                                          |
| team_name     | company City GmbH                     |
| platform      | ios                                           |
+---------------+-----------------------------------------------+

[14:31:15]: Starting login with user 'iosdev@company.com'
[14:31:17]: Successfully logged in
[14:31:25]: Successfully generated CERT_ID which was imported to the local machine.
[14:31:28]: Verifying the certificate is properly installed locally...
[14:31:28]: Successfully installed certificate CERT_ID

+-------------------------------------+-------------------------------------+
|                          Summary for sigh 2.48.0                          |
+-------------------------------------+-------------------------------------+
| app_identifier                      | com.company.app.NewProject |
| username                            | iosdev@company.com           |
| force                               | true                                |
| cert_id                             | CERT_ID                          |
| provisioning_name                   | match AdHoc                         |
|                                     | com.company.app.NewProject |
| ignore_profiles_with_different_name | true                                |
| team_id                             | TEAMID                          |
| platform                            | ios                                 |
| adhoc                               | true                                |
| development                         | false                               |
| skip_install                        | false                               |
| team_name                           | company City GmbH           |
| skip_fetch_profiles                 | false                               |
| skip_certificate_verification       | false                               |
+-------------------------------------+-------------------------------------+

[14:31:28]: Starting login with user 'iosdev@company.com'
[14:31:30]: Successfully logged in
[14:31:30]: Fetching profiles...
[14:31:30]: Verifying certificates...
[14:31:30]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
[14:31:32]: Creating new provisioning profile for 'com.company.app.NewProject' with name 'match AdHoc com.company.app.NewProject' for 'ios' platform
[14:31:34]: Downloading provisioning profile...
[14:31:35]: Successfully downloaded provisioning profile...
[14:31:35]: Installing provisioning profile...
/var/folders/lr/5vbfxcnd0vd3nm4djf1py7ycg0qddv/T/d20170726-71422-ox9juw/profiles/adhoc/AdHoc_com.company.app.NewProject.mobileprovision
[14:31:35]: Installing provisioning profile...
[14:31:36]: 🔒  Successfully encrypted certificates repo
[14:31:36]: Pushing changes to remote git repo...

+---------------------+------------------------------+------------------------------+
|                          Installed Provisioning Profile                           |
+---------------------+------------------------------+------------------------------+
| Parameter           | Environment Variable         | Value                        |
+---------------------+------------------------------+------------------------------+
| App Identifier      |                              | com.company.app.iOSMaste  |
|                     |                              | rProject                     |
| Type                |                              | adhoc                        |
| Platform            |                              | ios                          |
| Profile UUID        | sigh_com.company.app.iOS  | 95b7ab21-a77e-416a-a462-28b  |
|                     | MasterProject_adhoc          | 1297c6a57                    |
| Profile Name        | sigh_com.company.app.iOS  | match AdHoc                  |
|                     | MasterProject_adhoc_profile  | com.company.app.iOSMaste  |
|                     | -name                        | rProject                     |
| Profile Path        | sigh_com.company.app.iOS  | /Users/user/Library/Mobile  |
|                     | MasterProject_adhoc_profile  | Device/Provisioning          |
|                     | -path                        | Profiles/95b7ab21-a77e-416a  |
|                     |                              | -a462-28b1297c6a57.mobilepr  |
|                     |                              | ovision                      |
| Development Team ID | sigh_com.company.app.iOS  | TEAMID                   |
|                     | MasterProject_adhoc_team-id  |                              |
+---------------------+------------------------------+------------------------------+

[14:31:37]: All required keys, certificates and provisioning profiles are installed 🙌
[14:31:37]: Setting Provisioning Profile type to 'ad-hoc'
[14:31:37]: Cruising back to lane 'ios setup' 🚘
[14:31:37]: Lane setup completed successfully.

+------+------------------------------+-------------+
|                 fastlane summary                  |
+------+------------------------------+-------------+
| Step | Action                       | Time (in s) |
+------+------------------------------+-------------+
| 1    | Verifying required fastlane  | 0           |
|      | version                      |             |
| 2    | default_platform             | 0           |
| 3    | Switch to ios create_app_id  | 0           |
|      | lane                         |             |
| 4    | produce                      | 4           |
| 5    | Switch to ios sync_signing   | 0           |
|      | lane                         |             |
| 6    | match                        | 24          |
+------+------------------------------+-------------+

[14:31:37]: fastlane.tools finished successfully 🎉
</details>
### Environment
<details><summary>Output/Log</summary>

```


Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `length' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60length'%20for%20nil+repo:fastlane/fastlane


```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.4.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.9.2                           |
| Installation Source         | ~/.gem/bin/fastlane                         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.4.0/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | de_DE.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/blob/master/fastlane/Fastfile
#
# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`
#
# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.47.0"

default_platform :ios

desc "Print guide to configure new project"
lane :helpme do
	puts ""
	puts "***********************************************************************".cyan
	puts "*                                                                     *".cyan
	puts "* - Define values in 'fastlane/.env' or 'fastlane/.env.default' files *".cyan
	puts "* - Enable Apple Generic Versioning by follow the guide at:           *".cyan
	puts "*   https://developer.apple.com/library/content/qa/qa1827/_index.html *".cyan
	puts "* - Execute `fastlane setup`                                          *".cyan
	puts "*                                                                     *".cyan
	puts "***********************************************************************".cyan
	puts ""
end

platform :ios do
	desc "Create a new App ID on the Developer Portal"
	lane :setup do
		create_app_id
		sync_signing
	end

	desc "Configure, compile, analyze and deploy app"
	lane :build do
		prepare
		compile
		analyze
		deploy
	end

	desc "Configure app"
	lane :prepare do
		update_build_number
		install_cocoapods
		sync_signing
	end

	# https://docs.fastlane.tools/actions/#gym
	desc "Compile and sign app"
	lane :compile do
		disable_automatic_signing
		gym
	end

	desc "Run tests and static analizers"
	lane :analyze do
		unit_tests
	end

	desc "Deploy app"
	lane :deploy do
		# NOP
	end

	desc "Clean after build"
	lane :clean do
		Dir.chdir(ENV["PWD"]) do
			sh("rm -fr #{ENV['GYM_OUTPUT_DIRECTORY']}")
			sh("rm -fr #{ENV['SCAN_OUTPUT_DIRECTORY']}")
			sh("git co -- .")
		end
	end

	# https://docs.fastlane.tools/actions/#increment_build_number
	desc "Update build number"
	private_lane :update_build_number do
		project = ENV["PROJECT"]
		buildNumber = sh("date '+%Y%m%d%H%M'")

		increment_build_number(
			xcodeproj: project,
			build_number: buildNumber
		)
	end

	# https://docs.fastlane.tools/actions/#cocoapods
	desc "Install cocoapods"
	private_lane :install_cocoapods do
		podFile = ENV["POD_FILE"]

		cocoapods(
			podfile: podFile
		)
	end

	# https://docs.fastlane.tools/actions/#testing
	desc "Runs all unit tests"
	private_lane :unit_tests do
		scan
	end

	# https://docs.fastlane.tools/actions/#match
	desc "Sync certificates and provisioning profiles"
	private_lane :sync_signing do
		match
	end

	desc "Disable automatic code signing"
	private_lane :disable_automatic_signing do
		project = ENV["PROJECT"]
		teamId = ENV["TEAM_ID"]

		disable_automatic_code_signing(
			path: project,
			team_id: teamId
		)
	end

	desc "Enable automatic code signing"
	private_lane :enable_automatic_signing do
		project = ENV["PROJECT"]
		teamId = ENV["TEAM_ID"]

		enable_automatic_code_signing(
			path: project,
			team_id: teamId
		)
	end

	# https://docs.fastlane.tools/actions/#produce
	desc "Generate a new App ID on the Developer Portal"
	private_lane :create_app_id do
		identifiers = ENV["APP_IDENTIFIERS"].split(",")
		names = ENV["APP_NAMES"].split(",")

		if names.count != 0 && identifiers.count != names.count
			raise ArgumentError, "The number of identifiers does not match the number of names"
		end

		identifiers.each_index do |index|
			produce(
				app_identifier: identifiers.at(index).strip,
				app_name: names.at(index).strip
			)
		end
	end

	# desc "Generate release notes"
	# private_lane :release_notes do
	# get_version_number
	# changelog = changelog_from_git_commits
	# version_number = lane_context[SharedValues::VERSION_NUMBER]
	# {}"Release notes #{version_number} Automatic build (Last 5 PR):\n#{changelog}"
	# end

	############################# PRE ###############################

	# This block is executed before every action of all platforms
	before_all do |lane|
		puts "Lane #{lane} started."
	end

	############################# POST ##############################

	# This block is called, only if the executed lane was successful
	after_all do |lane|
		puts "Lane #{lane} completed successfully."
	end

	error do |lane, _exception|
		puts "Lane #{lane} failed to complete.".red
	end
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
#
# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`
#
# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# The bundle identifier of the app
app_identifier ENV["APP_IDENTIFIERS"]

# Shared Apple Developer Portal account
apple_id ENV["APPLE_ID"]

# Apple Developer Portal team
team_name ENV["TEAM_NAME"]

# Developer Portal Team ID
team_id ENV["TEAM_ID"]
```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.48.0  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.1.0        |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.11.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170223 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| json                      | 2.0.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| io-console                | 0.4.6        |
| openssl                   | 2.0.2        |
</details>

*generated on:* **2017-07-26**
</details>

[New project](https://prnt.sc/g0l33m)
[Existing project](https://prnt.sc/g0l3e8)

**EDIT 2017-07-27**
I have updated fastlane to 2.49.0 and removed new created certificate and pp from dev portal and local machine -> old project work again.
Tried to specify SIGH_CERT_ID but it creates new certificate again and old project does not compile.

**Comment by ohayon** [comment]
Hey @gtatgh - apologies for the issues you are having! Did you make sure that you are doing `Dotenv.load` at the top of your Fastfile if you are storing data in there?

**Comment by gtatgh** [comment]
Hello @ohayon, no I don't make `Dotenv.load`. If I really understand [this](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Advanced.md#environment-variables) description fastlane do it automatically. I will try it.

**Comment by gtatgh** [comment]
I have added verbose match log.
<details><summary>match log</summary>
11:44 user:iOSProject$ fastlane match adhoc --git_url 'ssh://Bamboo-Daemon@xyz-stash:7999/gad/ios_certificates.git' --git_branch 'com.company.app.iOSProject' --app_identifier 'com.company.app.iOSProject' --username 'iosdev@company.com' --team_id 'TEAM_ID' --platform 'ios' --verbose true
[11:45:07]: For more information, check out https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------

+-----------------------+-------------------------------------------------------------------+
|                                 Summary for match 2.49.0                                  |
+-----------------------+-------------------------------------------------------------------+
| verbose               | true                                                              |
| git_url               | ssh://Bamboo-Daemon@xyz-stash:7999/gad/ios_certificates.git |
| git_branch            | com.company.app.iOSProject                               |
| app_identifier        | com.company.app.iOSProject                               |
| username              | iosdev@company.com                                         |
| team_id               | TEAM_ID                                                        |
| platform              | ios                                                               |
| type                  | adhoc                                                             |
| keychain_name         | login.keychain                                                    |
| readonly              | false                                                             |
| force                 | false                                                             |
| skip_confirmation     | false                                                             |
| shallow_clone         | false                                                             |
| clone_branch_directly | false                                                             |
| force_for_new_devices | false                                                             |
| skip_docs             | false                                                             |
+-----------------------+-------------------------------------------------------------------+

INFO [2017-07-28 11:45:09.85]: Cloning remote git repo...
INFO [2017-07-28 11:45:09.85]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
INFO [2017-07-28 11:45:09.85]: $ GIT_TERMINAL_PROMPT=0 git clone 'ssh://Bamboo-Daemon@xyz-stash:7999/gad/ios_certificates.git' '/var/folders/lr/5vbfxcnd0vd3nm4djf1py7ycg0qddv/T/d20170728-56683-rbafkj'
INFO [2017-07-28 11:45:09.85]: ▸ Cloning into '/var/folders/lr/5vbfxcnd0vd3nm4djf1py7ycg0qddv/T/d20170728-56683-rbafkj'...
INFO [2017-07-28 11:45:10.14]: ▸ remote: Counting objects: 610, done.
INFO [2017-07-28 11:45:10.30]: ▸ remote: Compressing objects: 100% (534/534), done.
INFO [2017-07-28 11:45:10.40]: ▸ remote: Total 610 (delta 58), reused 270 (delta 28)
INFO [2017-07-28 11:45:10.40]: ▸ Receiving objects: 100% (610/610), 1.23 MiB | 0 bytes/s, done.
INFO [2017-07-28 11:45:10.40]: ▸ Resolving deltas: 100% (58/58), done.
INFO [2017-07-28 11:45:10.41]: ▸ Checking connectivity... done.
INFO [2017-07-28 11:45:10.41]: $ git branch --list origin/com.company.app.iOSProject --no-color -r
INFO [2017-07-28 11:45:10.42]: Checking out branch com.company.app.iOSProject...
INFO [2017-07-28 11:45:10.42]: $ git checkout --orphan com.company.app.iOSProject
INFO [2017-07-28 11:45:10.42]: ▸ Switched to a new branch 'com.company.app.iOSProject'
INFO [2017-07-28 11:45:10.42]: $ git reset --hard
INFO [2017-07-28 11:45:10.43]: 🔓  Successfully decrypted certificates repo
INFO [2017-07-28 11:45:10.45]: Verifying that the certificate and profile are still valid on the Dev Portal...
WARN [2017-07-28 11:45:13.65]: Couldn't find a valid code signing identity in the git repo for distribution... creating one for you now
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------

+---------------+--------------------------------------------------+
|                     Summary for cert 2.49.0                      |
+---------------+--------------------------------------------------+
| development   | false                                            |
| force         | true                                             |
| username      | iosdev@company.com                        |
| team_id       | TEAM_ID                                       |
| keychain_path | /Users/user/Library/Keychains/login.keychain-db |
| platform      | ios                                              |
+---------------+--------------------------------------------------+

INFO [2017-07-28 11:45:13.66]: Starting login with user 'iosdev@company.com'
INFO [2017-07-28 11:45:15.38]: Successfully logged in
INFO [2017-07-28 11:45:17.62]: $ security import /var/folders/lr/5vbfxcnd0vd3nm4djf1py7ycg0qddv/T/d20170728-56683-rbafkj/certs/distribution/8RYGF4SEHX.p12 -k '/Users/user/Library/Keychains/login.keychain-db' -P '' -T /usr/bin/codesign -T /usr/bin/security
INFO [2017-07-28 11:45:17.85]: ▸ 1 key imported.
INFO [2017-07-28 11:45:17.85]: $ security set-key-partition-list -S apple-tool:,apple: -k '' /Users/user/Library/Keychains/login.keychain-db
security: SecKeychainItemSetAccessWithPassword: The user name or passphrase you entered is not correct.
INFO [2017-07-28 11:45:20.74]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:20.74]: ▸ version: 512
INFO [2017-07-28 11:45:20.74]: ▸ class: 0x0000000F 
INFO [2017-07-28 11:45:20.74]: ▸ attributes:
INFO [2017-07-28 11:45:20.74]: ▸     0x00000000 <uint32>=0x0000000F 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000001 <blob>="<key>"
INFO [2017-07-28 11:45:20.74]: ▸     0x00000002 <blob>=<NULL>
INFO [2017-07-28 11:45:20.74]: ▸     0x00000003 <uint32>=0x00000001 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000004 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000005 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000006 <blob>=0xB26BD07839DCAA305A996AA647B707EF22F778CF  "\262k\320x9\334\2520Z\231j\246G\267\007\357"\367x\317"
INFO [2017-07-28 11:45:20.74]: ▸     0x00000007 <blob>=<NULL>
INFO [2017-07-28 11:45:20.74]: ▸     0x00000008 <blob>=0x7B38373139316361322D306663392D313164342D383439612D3030303530326235323132327D00  "{87191ca2-0fc9-11d4-849a-000502b52122}\000"
INFO [2017-07-28 11:45:20.74]: ▸     0x00000009 <uint32>=0x0000002A  "\000\000\000*"
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000A <uint32>=0x00000800 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000B <uint32>=0x00000800 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000C <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000D <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000E <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000000F <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000010 <uint32>=0x00000001 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000011 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000012 <uint32>=0x00000001 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000013 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000014 <uint32>=0x00000001 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000015 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000016 <uint32>=0x00000001 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000017 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000018 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x00000019 <uint32>=0x00000000 
INFO [2017-07-28 11:45:20.74]: ▸     0x0000001A <uint32>=0x00000000 
...
INFO [2017-07-28 11:45:25.26]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:25.26]: ▸ version: 512
INFO [2017-07-28 11:45:25.26]: ▸ class: 0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸ attributes:
INFO [2017-07-28 11:45:25.26]: ▸     0x00000000 <uint32>=0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000001 <blob>="iMessage Signing Key"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000002 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000003 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000004 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000005 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000006 <blob>=0xED43E458C0719AD6FEE0DCBE6E64FF5CB37C37CC  "\355C\344X\300q\232\326\376\340\334\276nd\377\134\263|7\314"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000007 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000008 <blob>=0x7B38373139316361322D306663392D313164342D383439612D3030303530326235323132327D00  "{87191ca2-0fc9-11d4-849a-000502b52122}\000"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000009 <uint32>=0x00000049  "\000\000\000I"
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000A <uint32>=0x00000100 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000B <uint32>=0x00000100 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000C <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000D <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000E <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000F <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000010 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000011 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000012 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000013 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000014 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000015 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000016 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000017 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000018 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000019 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000001A <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:25.26]: ▸ version: 512
INFO [2017-07-28 11:45:25.26]: ▸ class: 0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸ attributes:
INFO [2017-07-28 11:45:25.26]: ▸     0x00000000 <uint32>=0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000001 <blob>="iMessage Encryption Key"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000002 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000003 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000004 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000005 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000006 <blob>=0x6F910BC677E90E25FE17F702C1A5D278B15EF32D  "o\221\013\306w\351\016%\376\027\367\002\301\245\322x\261^\363-"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000007 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000008 <blob>=0x7B38373139316361322D306663392D313164342D383439612D3030303530326235323132327D00  "{87191ca2-0fc9-11d4-849a-000502b52122}\000"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000009 <uint32>=0x0000002A  "\000\000\000*"
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000A <uint32>=0x00000500 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000B <uint32>=0x00000500 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000C <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000D <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000E <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000F <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000010 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000011 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000012 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000013 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000014 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000015 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000016 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000017 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000018 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000019 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000001A <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:25.26]: ▸ version: 512
INFO [2017-07-28 11:45:25.26]: ▸ class: 0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸ attributes:
INFO [2017-07-28 11:45:25.26]: ▸     0x00000000 <uint32>=0x00000010 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000001 <blob>="iMessage Signing Key"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000002 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000003 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000004 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000005 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000006 <blob>=0xD1F07D3B65C72E11C08C2C2382FD8B1DB3C47CEC  "\321\360};e\307.\021\300\214,#\202\375\213\035\263\304|\354"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000007 <blob>=<NULL>
INFO [2017-07-28 11:45:25.26]: ▸     0x00000008 <blob>=0x7B38373139316361322D306663392D313164342D383439612D3030303530326235323132327D00  "{87191ca2-0fc9-11d4-849a-000502b52122}\000"
INFO [2017-07-28 11:45:25.26]: ▸     0x00000009 <uint32>=0x00000049  "\000\000\000I"
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000A <uint32>=0x00000100 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000B <uint32>=0x00000100 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000C <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000D <blob>=0x0000000000000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000E <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000000F <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000010 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000011 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000012 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000013 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000014 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000015 <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000016 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000017 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000018 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x00000019 <uint32>=0x00000000 
INFO [2017-07-28 11:45:25.26]: ▸     0x0000001A <uint32>=0x00000001 
INFO [2017-07-28 11:45:25.26]: Successfully generated 8RYGF4SEHX which was imported to the local machine.
INFO [2017-07-28 11:45:25.26]: $ security list-keychains -d user
INFO [2017-07-28 11:45:25.27]: ▸     "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:25.27]: $ security find-certificate -c 'Apple Worldwide Developer Relations Certification Authority' /Users/user/Library/Keychains/login.keychain-db
INFO [2017-07-28 11:45:25.29]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:25.29]: ▸ version: 512
INFO [2017-07-28 11:45:25.29]: ▸ class: 0x80001000 
INFO [2017-07-28 11:45:25.29]: ▸ attributes:
INFO [2017-07-28 11:45:25.29]: ▸     "alis"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-28 11:45:25.29]: ▸     "cenc"<uint32>=0x00000003 
INFO [2017-07-28 11:45:25.29]: ▸     "ctyp"<uint32>=0x00000001 
INFO [2017-07-28 11:45:25.29]: ▸     "hpky"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-28 11:45:25.29]: ▸     "issu"<blob>=0x3062310B300906035504061302555331133011060355040A130A4150504C4520494E432E31263024060355040B131D4150504C452043455254494649434154494F4E20415554484F52495459311630140603550403130D4150504C4520524F4F54204341  "0b1\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\023\012APPLE INC.1&0$\006\003U\004\013\023\035APPLE CERTIFICATION AUTHORITY1\0260\024\006\003U\004\003\023\015APPLE ROOT CA"
INFO [2017-07-28 11:45:25.29]: ▸     "labl"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-28 11:45:25.29]: ▸     "skid"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-28 11:45:25.29]: ▸     "snbr"<blob>=0x19 
INFO [2017-07-28 11:45:25.29]: ▸     "subj"<blob>=0x308196310B300906035504061302555331133011060355040A0C0A4170706C6520496E632E312C302A060355040B0C234170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E733144304206035504030C3B4170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E732043657274696669636174696F6E20417574686F72697479  "0\201\2261\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\014\012Apple Inc.1,0*\006\003U\004\013\014#Apple Worldwide Developer Relations1D0B\006\003U\004\003\014;Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-28 11:45:25.67]: Verifying the certificate is properly installed locally...
INFO [2017-07-28 11:45:25.67]: Successfully installed certificate 8RYGF4SEHX
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------
Successfully loaded Appfile at path '/Users/user/projects/ios/xyz/iOSProject/fastlane/Appfile'
-------

+-------------------------------------+-------------------------------------------------+
|                                Summary for sigh 2.49.0                                |
+-------------------------------------+-------------------------------------------------+
| app_identifier                      | com.company.app.iOSProject             |
| username                            | iosdev@company.com                       |
| force                               | true                                            |
| cert_id                             | 8RYGF4SEHX                                      |
| provisioning_name                   | match AdHoc com.company.app.iOSProject |
| ignore_profiles_with_different_name | true                                            |
| team_id                             | TEAM_ID                                      |
| platform                            | ios                                             |
| adhoc                               | true                                            |
| development                         | false                                           |
| skip_install                        | false                                           |
| skip_fetch_profiles                 | false                                           |
| skip_certificate_verification       | false                                           |
+-------------------------------------+-------------------------------------------------+

INFO [2017-07-28 11:45:25.93]: Starting login with user 'iosdev@company.com'
INFO [2017-07-28 11:45:28.16]: Successfully logged in
INFO [2017-07-28 11:45:28.16]: Fetching profiles...
INFO [2017-07-28 11:45:28.96]: Verifying certificates...
WARN [2017-07-28 11:45:28.96]: No existing profiles found, that match the certificates you have installed locally! Creating a new provisioning profile for you
INFO [2017-07-28 11:45:30.82]: $ security list-keychains -d user
INFO [2017-07-28 11:45:30.83]: ▸     "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:30.83]: $ security find-certificate -c 'Apple Worldwide Developer Relations Certification Authority' /Users/user/Library/Keychains/login.keychain-db
INFO [2017-07-28 11:45:30.85]: ▸ keychain: "/Users/user/Library/Keychains/login.keychain-db"
INFO [2017-07-28 11:45:30.85]: ▸ version: 512
INFO [2017-07-28 11:45:30.85]: ▸ class: 0x80001000 
INFO [2017-07-28 11:45:30.85]: ▸ attributes:
INFO [2017-07-28 11:45:30.85]: ▸     "alis"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-28 11:45:30.85]: ▸     "cenc"<uint32>=0x00000003 
INFO [2017-07-28 11:45:30.85]: ▸     "ctyp"<uint32>=0x00000001 
INFO [2017-07-28 11:45:30.85]: ▸     "hpky"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-28 11:45:30.85]: ▸     "issu"<blob>=0x3062310B300906035504061302555331133011060355040A130A4150504C4520494E432E31263024060355040B131D4150504C452043455254494649434154494F4E20415554484F52495459311630140603550403130D4150504C4520524F4F54204341  "0b1\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\023\012APPLE INC.1&0$\006\003U\004\013\023\035APPLE CERTIFICATION AUTHORITY1\0260\024\006\003U\004\003\023\015APPLE ROOT CA"
INFO [2017-07-28 11:45:30.85]: ▸     "labl"<blob>="Apple Worldwide Developer Relations Certification Authority"
INFO [2017-07-28 11:45:30.85]: ▸     "skid"<blob>=0x88271709A9B618608BECEBBAF64759C55254A3B7  "\210'\027\011\251\266\030`\213\354\353\272\366GY\305RT\243\267"
INFO [2017-07-28 11:45:30.85]: ▸     "snbr"<blob>=0x19 
INFO [2017-07-28 11:45:30.85]: ▸     "subj"<blob>=0x308196310B300906035504061302555331133011060355040A0C0A4170706C6520496E632E312C302A060355040B0C234170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E733144304206035504030C3B4170706C6520576F726C647769646520446576656C6F7065722052656C6174696F6E732043657274696669636174696F6E20417574686F72697479  "0\201\2261\0130\011\006\003U\004\006\023\002US1\0230\021\006\003U\004\012\014\012Apple Inc.1,0*\006\003U\004\013\014#Apple Worldwide Developer Relations1D0B\006\003U\004\003\014;Apple Worldwide Developer Relations Certification Authority"
WARN [2017-07-28 11:45:31.83]: Creating new provisioning profile for 'com.company.app.iOSProject' with name 'match AdHoc com.company.app.iOSProject' for 'ios' platform
WARN [2017-07-28 11:45:33.58]: Downloading provisioning profile...
INFO [2017-07-28 11:45:34.84]: Successfully downloaded provisioning profile...
INFO [2017-07-28 11:45:34.89]: Installing provisioning profile...
/var/folders/lr/5vbfxcnd0vd3nm4djf1py7ycg0qddv/T/d20170728-56683-rbafkj/profiles/adhoc/AdHoc_com.company.app.iOSProject.mobileprovision
INFO [2017-07-28 11:45:34.93]: Installing provisioning profile...
WARN [2017-07-28 11:45:35.80]: Setting environment variable 'sigh_com.company.app.iOSProject_adhoc' to '9a006a4c-1f09-4e71-91c1-acec86633c07'
WARN [2017-07-28 11:45:35.80]: Setting environment variable 'sigh_com.company.app.iOSProject_adhoc_team-id' to 'TEAM_ID'
WARN [2017-07-28 11:45:35.80]: Setting environment variable 'sigh_com.company.app.iOSProject_adhoc_profile-name' to 'match AdHoc com.company.app.iOSProject'
WARN [2017-07-28 11:45:35.80]: Setting environment variable 'sigh_com.company.app.iOSProject_adhoc_profile-path' to '/Users/user/Library/MobileDevice/Provisioning Profiles/9a006a4c-1f09-4e71-91c1-acec86633c07.mobileprovision'
INFO [2017-07-28 11:45:35.84]: 🔒  Encrypted '8RYGF4SEHX.cer'
INFO [2017-07-28 11:45:35.87]: 🔒  Encrypted '8RYGF4SEHX.p12'
INFO [2017-07-28 11:45:35.90]: 🔒  Encrypted 'AdHoc_com.company.app.iOSProject.mobileprovision'
INFO [2017-07-28 11:45:35.90]: 🔒  Successfully encrypted certificates repo
INFO [2017-07-28 11:45:35.90]: Pushing changes to remote git repo...
INFO [2017-07-28 11:45:35.90]: $ git add -A
INFO [2017-07-28 11:45:35.91]: $ git commit -m \[fastlane\]\ Updated\ adhoc\ and\ platform\ ios
INFO [2017-07-28 11:45:35.92]: ▸ [com.company.app.iOSProject (root-commit) 74760ed] [fastlane] Updated adhoc and platform ios
INFO [2017-07-28 11:45:35.92]: ▸ 5 files changed, 300 insertions(+)
INFO [2017-07-28 11:45:35.92]: ▸ create mode 100644 README.md
INFO [2017-07-28 11:45:35.92]: ▸ create mode 100644 certs/distribution/8RYGF4SEHX.cer
INFO [2017-07-28 11:45:35.92]: ▸ create mode 100644 certs/distribution/8RYGF4SEHX.p12
INFO [2017-07-28 11:45:35.92]: ▸ create mode 100644 match_version.txt
INFO [2017-07-28 11:45:35.92]: ▸ create mode 100644 profiles/adhoc/AdHoc_com.company.app.iOSProject.mobileprovision
INFO [2017-07-28 11:45:35.92]: $ GIT_TERMINAL_PROMPT=0 git push origin com.company.app.iOSProject
INFO [2017-07-28 11:45:36.07]: ▸ Counting objects: 11, done.
INFO [2017-07-28 11:45:36.07]: ▸ Delta compression using up to 8 threads.
INFO [2017-07-28 11:45:36.07]: ▸ Compressing objects: 100% (8/8), done.
INFO [2017-07-28 11:45:36.07]: ▸ Writing objects: 100% (11/11), 12.52 KiB | 0 bytes/s, done.
INFO [2017-07-28 11:45:36.07]: ▸ Total 11 (delta 0), reused 1 (delta 0)
INFO [2017-07-28 11:45:36.40]: ▸ remote:
INFO [2017-07-28 11:45:36.40]: ▸ remote: Create pull request for com.company.app.iOSProject:
INFO [2017-07-28 11:45:36.40]: ▸ remote:   http://xyz-bitbucket/projects/GAD/repos/ios_certificates/compare/commits?sourceBranch=refs/heads/com.company.app.iOSProject
INFO [2017-07-28 11:45:36.40]: ▸ remote:
INFO [2017-07-28 11:45:36.41]: ▸ To ssh://Bamboo-Daemon@xyz-stash:7999/gad/ios_certificates.git
INFO [2017-07-28 11:45:36.41]: ▸ * [new branch]      com.company.app.iOSProject -> com.company.app.iOSProject

+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
|                                                          Installed Provisioning Profile                                                           |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
| Parameter           | Environment Variable                                        | Value                                                         |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+
| App Identifier      |                                                             | com.company.app.iOSProject                           |
| Type                |                                                             | adhoc                                                         |
| Platform            |                                                             | ios                                                           |
| Profile UUID        | sigh_com.company.app.iOSProject_adhoc              | 9a006a4c-1f09-4e71-91c1-acec86633c07                          |
| Profile Name        | sigh_com.company.app.iOSProject_adhoc_profile-name | match AdHoc com.company.app.iOSProject               |
| Profile Path        | sigh_com.company.app.iOSProject_adhoc_profile-path | /Users/user/Library/MobileDevice/Provisioning                |
|                     |                                                             | Profiles/9a006a4c-1f09-4e71-91c1-acec86633c07.mobileprovision |
| Development Team ID | sigh_com.company.app.iOSProject_adhoc_team-id      | TEAM_ID                                                    |
+---------------------+-------------------------------------------------------------+---------------------------------------------------------------+

INFO [2017-07-28 11:45:36.42]: All required keys, certificates and provisioning profiles are installed 🙌
</details>

**Comment by gtatgh** [comment]
If I run
`fastlane sigh --adhoc true --app_identifier 'com.company.app.iosproject' --username 'iOSDev@company.com' --team_id 'TEAM_ID' --cert_id 'CERT_ID' --platform 'ios'`
then right provisioning profiles will be cretaed. But I can't import them to the match without additional cost. Well, where is the mistake I did? I do not think that this is a bug in faslane match, otherwise this issue was already reported by someone.

**Comment by gtatgh** [comment]
Hi @ohayon,
I think I have found the problem. We use match with different branches for each project.
- Customer A: certificate A -> project A -> branch A
- Customer B: certificate B -> project B -> branch B
- Customer C: certificate C -> project C -> branch C
- Customer C: *should be certificate C*, but **certificate D will be created** -> project D -> branch D
This strategy is useful if the maintaince for project A (as example) ends, we delete the branch A. And remove certs and pp from a developer machine. The other point is *confidentiality*. If developer 1 works on project A and developer 2 on project B and C. So the developers don't need certificates from other projects to be installed.
What is the best approach to use `match` with these points?

**Comment by ohayon** [comment]
@gtatgh - I'm not _totally_ sure I understand the issue you are describing. If I am understanding correctly, different projects are using the same repo for your certificates that are managed with _match_, just on different branches? I am not sure that _match_ supports that workflow out of the box. Is it possible that you could set up different repos for each project? Let me know if that makes sense or if I am misunderstanding the issue. I hope we can figure this out and get you unblocked! 🐙

**Comment by gtatgh** [comment]
@ohayon Yes, you understand that problem correct. We are using the same repo for our apps that are managed with *match*, just on different branches. Also each app has its own branch (branch name is equal to the project id). Now I'm understanding too that match does not supports that workflow. But is it possible to reqest this feature?

**Comment by ohayon** [comment]
@gtatgh Could you elaborate on what you mean as far as checking certificates in the developer portal? Like comparing the bundle ID associated with each profile to make sure it matches the app's bundle ID? 😄

**Comment by gtatgh** [comment]
@ohayon I mean if I have already a certificate and provisioning profile created by *match* and stored in **certificate repo** under branch with name A. This branch is used by *match* to sync certificates and provisioning profiles for project A. This works fine. Now I create new project B and I want that *match* uses already **existing certificate**, but instead of using **certificate repo -> branch A** I want to use **certificate repo -> branch B** to sync certificates and provisioning profiles for newly created project B.
But, how I learned this fails, because match create new branch B in **certificate repo** and look inside but find no certificate. Right, the certificate is stored in **certificate repo** under branch A and instead of looking in Apples developer portal for existing certificates (for this purpose new option `MATCH_CERT_ID` may be helpful), *match* create new certificate and store it in **certificate repo** under branch B. As result the new created certificate breaks signing of project A.

### Now a feature request
Before creating new certificate *match* should look in Apples developer portal for existing certificate specified by new option `MATCH_CERT_ID` and if found download them and store under specified branch. Followed an exsample of command, maybe this help to understand what I mean:

`fastlane match appstore --git_url 'ssh://.../ios_certificates.git' --git_branch 'project.bundle.id' --app_identifier 'project.bundle.id' --username 'iOSDev@company.com' --team_id 'TEAM_ID' --platform 'ios' --match_cert_id 'USE_MY_EXISTING_CERT_ID_TO_DOWNLOAD_RIGHT_CERT_FROM_APPLES_DEVELOPER_PORTAL'`

**Comment by ohayon** [comment]
Hmm, This is an interesting use case! Would you want to create a PR for this and we can see if it would be feasible to integrate into _match_? 🐙

**Comment by gtatgh** [comment]
@ohayon Yes, I can create a PR, but need some help.

I can dowload certificate, but failed on problem creating *personal information exchange (.p12)*.

Is the following piece of code  to create *p12* right? If so, what is the password? Is it the apple's developer portal password? What is the type (Is it 'distribution', 'development')?
`p12 = OpenSSL::PKCS12.create('password', 'type', x509_certificate.public_key, x509_certificate)`

How can I check if created exchange is valid? For the line `result = x509_certificate.check_private_key(p12)` I become following error 
`undefined method private?' for #<OpenSSL::PKCS12:0x007fd0a5b0ea00>`

**Comment by gtatgh** [comment]
I have created a [branch](https://github.com/fastlane/fastlane/compare/master...gtatgh:9865) for this, but need some help.

1. What is the value for `password`?
`p12 = OpenSSL::PKCS12.create('password', 'type', x509_certificate.public_key, x509_certificate)`
2. If I write *p12* object to file, it will be written binary. What do I wrong?
`File.write(p12_path, p12.to_der)`

**Comment by nicolasbraun** [comment]
Just stepping because we have kind of the same usage avec `match`. 
Why don't you simply use a branch per Apple Developer Team ?

**Comment by gtatgh** [comment]
If you do not support some project, you need to check a branch out, delete provisioning profiles and commit it back, instead of just delete a branch.

**Comment by gtatgh** [comment]
If you do not support some project, you need to check a branch out, delete provisioning profiles and commit it back, instead of just delete a branch.
